calculator/main.py

```diff
@@ -25,9 +26,21 @@ def perform_operation(calc, operation, params):
     :param operation: String with the operation name. ie: 'add'
     :param params: Tuple containing the list of nums to operate with.
                    ie: (1, 2, 3, 4.5, -2)
+                   
     """
-    pass
-
+    if operation not in calc['operations'].keys():
+        raise InvalidOperation("Given operation is invalid.")
+
+ 
+    for element in params:
+        if type(element) == str:
```

Comment:
Like Santiago said, it might be easier in this case to check for the parameters you can use. For example: You could technically pass a Bool but get no desired InvalidParams error